File: services/shared/schemas/company.ts
import { z } from "zod";

// A company is the top-level tenant.
// `type='agency'` rolls up multiple `type='client'` companies via parent_company_id.

export const CompanyTypeSchema = z.enum(["agency", "client", "in_house", "solo"]);
export type CompanyType = z.infer<typeof CompanyTypeSchema>;

export const UiModeSchema = z.enum(["web2", "web3"]);
export type UiMode = z.infer<typeof UiModeSchema>;

export const CompanySchema = z.object({
  id: z.string().uuid(),
  name: z.string().min(1).max(120),
  type: CompanyTypeSchema,
  parentCompanyId: z.string().uuid().nullable().optional(),
  uiMode: UiModeSchema.default("web2"),
  brandKitId: z.string().uuid().nullable().optional(),
  activeRegimes: z.array(z.string()).default([]), // e.g. ['mica', 'fca']
  contextJson: z.record(z.unknown()).default({}),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
});
export type Company = z.infer<typeof CompanySchema>;

export const CompanyCreateSchema = CompanySchema.pick({
  name: true,
  type: true,
  parentCompanyId: true,
  uiMode: true,
});
export type CompanyCreate = z.infer<typeof CompanyCreateSchema>;

File: services/shared/schemas/company.py
"""Company schema — Python mirror of company.ts."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

CompanyType = Literal["agency", "client", "in_house", "solo"]
UiMode = Literal["web2", "web3"]


class Company(BaseModel):
    id: str
    name: str = Field(..., min_length=1, max_length=120)
    type: CompanyType
    parent_company_id: str | None = None
    ui_mode: UiMode = "web2"
    brand_kit_id: str | None = None
    active_regimes: list[str] = Field(default_factory=list)
    context_json: dict[str, object] = Field(default_factory=dict)
    created_at: datetime
    updated_at: datetime


class CompanyCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=120)
    type: CompanyType
    parent_company_id: str | None = None
    ui_mode: UiMode = "web2"
